### PR TITLE
Ability to connect using sessionid instead of username and password

### PR DESCRIPTION
**Pull request by jp-harvey**
Currently we require a user name and password to create a valid `Connection` object. With work currently being done to create an authenticator with Jupyter Hub, we need to be able to validate an existing OmniSci session. While we can build the Thrift transport manually in the authenticator (like pymapd does), it seems we'll likely need this in the future for something else we're integrating with so it makes more sense to make it a part of pymapd proper.

### CODE REVIEW

**Comment by jp-harvey** [comment]
@randyzwitch note I have not done any tests as it would require interacting with the live OmniSci instance and not sure exactly how to do that with our CI. lmk if this is required and point me in the right direction and happy to do so.

**Comment by andrewseidl** [comment]
https://github.com/omnisci/pymapd/blob/3422883656d938086785944f7feb0267319c6fd6/tests/test_connection.py#L10-L43

File: pymapd/connection.py [hunk 1]
```diff
@@ -169,7 +181,12 @@ def __init__(self,
                 raise
         self._client = Client(proto)
         try:
-            self._session = self._client.connect(user, password, dbname)
+            # If a sessionid was passed, we should validate it
+            if sessionid:
+                self._session = sessionid
+                self.get_tables()
```
Comment (randyzwitch) [inline pymapd/connection.py]: @jp-harvey Have you tried this with a failing sessionid to see what it throws? I just fixed the wrong port issue to give it a more descriptive suggestion on what to fix...depending on what an invalid sessionid throws, we might consider having a custom error message

File: tests/test_connection.py [hunk 2]
```diff
@@ -42,6 +43,18 @@ def test_bad_protocol(self, mock_client):
...
 
+    def test_session_logon_success(self):
+        conn = connect(user='mapd', password='HyperInteractive',
+                       host='localhost', dbname='mapd')
+        sessionid = conn._session
+        connnew = connect(sessionid=sessionid, host='localhost')
+        assert connnew._session == sessionid
+
+    def test_session_logon_failure(self):
+        sessionid = 'ILoveDancingOnTables'
```
Comment (randyzwitch) [inline tests/test_connection.py]: well played :100:

**Comment by randyzwitch** [comment]
LGTM, @jp-harvey just see the one comment about what exception is thrown if sessionid is invalid

File: pymapd/connection.py (hunk 1 above)
Comment (jp-harvey) [reply pymapd/connection.py]: Yep it's already pretty unambiguous in the context of a user trying to create a connection using a session: `pymapd.exceptions.Error: Session not valid.`

File: tests/test_connection.py (hunk 2 above)
Comment (jp-harvey) [reply tests/test_connection.py]: :-)

File: pymapd/connection.py (hunk 1 above)
Comment (randyzwitch) [reply pymapd/connection.py]: :+1:

**Pull request merged**